chore: tanstack query 설치

devjaeyoon opened the pull request:
### ✨ 이슈 번호

- #63 

### 📌 설명

tanstack query를 사용하기 위해 @tanstack/react-query, 개발 환경에 도움을 주는 @tanstack/react-query-devtools, tanstack query 코드 작성에 도움을 주는 @tanstack/eslint-plugin-query를 설치하고 적용한 Pull Request입니다.

### 📃 작업 사항

- [x] @tanstack/react-query 설치
- [x] @tanstack/react-query-devtools 설치 및 ReactQueryDevtools 적용
- [x] @tanstack/eslint-plugin-query 설치 및 플러그인 적용
- [x] QueryClient 및 QueryClientProvider 설정

### 💭 리뷰 사항 반영

- [x] 배포 시 `ReactQueryDevtools`가 제거될 수 있도록 `{import.meta.env.DEV && <ReactQueryDevtools initialIsOpen={false} />}` 작업

### ✅ Pull Request 체크 사항

- [x] 가장 최신 브랜치를 pull 하였습니다.
- [x] base 브랜치명을 확인하였습니다.
- [x] 코드 컨벤션을 모두 확인하였습니다.
- [x] 브랜치명을 확인하였습니다.

devjaeyoon committed:
chore: @tanstack/react-query 설치

devjaeyoon committed:
chore: @tanstack/eslint-plugin-query 설치

devjaeyoon committed:
chore: @tanstack/eslint-plugin-query eslint.config.js에 적용

devjaeyoon committed:
feat: QueryClient 및 QueryClientProivder 추가

devjaeyoon committed:
chore: @tanstack/react-query-devtools 설치

devjaeyoon committed:
feat: ReactQueryDevtools 추가

faber-h commented on src/App.jsx:
```diff
@@ -1,9 +1,18 @@
+  return (
+    <QueryClientProvider client={queryClient}>
+      <RouterProvider router={router} />
+      <ReactQueryDevtools initialIsOpen={false} />
```
tanstack query 설치를 잘해주셨네요!
다만, 배포 시에 `ReactQueryDevtools` 코드를 제거해야 된다는 주석을 달면 좋을 것 같아요~

devjaeyoon replied:
좋은 리뷰 감사드립니다!

해당 작업을
```
{import.meta.env.DEV && <ReactQueryDevtools initialIsOpen={false} />}
```
와 같은 코드를 작성하여 현재 코드에서는 `import.meta.env.DEV` 조건으로 감싸 `Devtools`는 개발 모드에서만 렌더링되도록 했고, `Vite`의 `tree shaking`과 `dead code elimination` 덕분에 배포 빌드에서는 관련 코드도 제거됩니다. 따라서 따로 제거하지 않아도 `production` 번들에 포함되지 않습니다!

항상 제가 작성한 코드에 대해 한 번 더 생각해 볼 수 있는 기회를 제공해주셔서 진심으로 감사드립니다.

devjaeyoon committed:
feat: ReactQueryDevtools를 개발 환경에서만 렌더링하도록 변경

- Vite의 import.meta.env.DEV를 사용해 Devtools를 개발 환경에서만 렌더링
- production 빌드에서 Devtools 코드가 번들에 포함되지 않도록 설정

faber-h reviewed (approved):
LGTM 🔥

Eun0713 reviewed (approved):
LGTM🌟
